From 10fb6b1c7b95b6f95c7a19f18cfd1798605016b4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20Kotyla?= <michal.kotyla@3mdeb.com>
Date: Fri, 15 Apr 2022 15:52:21 +0200
Subject: [PATCH] main.c: IOMMU flushing inifnity loop workaround

---
 main.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/main.c b/main.c
index d2c727a..d125fa6 100644
--- a/main.c
+++ b/main.c
@@ -341,9 +341,10 @@ static void iommu_setup(void)
 
 		iommu_load_device_table(iommu_cap, &iommu_done);
 		print("Flushing IOMMU cache");
-		while (!iommu_done) {
-			print(".");
-		}
+		/* Inifnity loop workaround */
+		//while (!iommu_done) {
+			//print(".");
+		//}
 		print("\nIOMMU set\n");
 	}
 
-- 
2.25.1

