From 8642d4bcf7afafb35210309633cdb7b9403dfed6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20Kotyla?= <michal.kotyla@3mdeb.com>
Date: Fri, 15 Apr 2022 17:36:11 +0200
Subject: [PATCH] main.c: IOMMU flushing inifnity loop workaround

---
 main.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/main.c b/main.c
index 763647c..dc4533c 100644
--- a/main.c
+++ b/main.c
@@ -379,9 +379,10 @@ static asm_return_t lz_linux(struct tpm *tpm, struct lz_tag_boot_linux *lz_tag)
 
 		iommu_load_device_table(iommu_cap, &iommu_done);
 		print("Flushing IOMMU cache");
-		while (!iommu_done) {
-			print(".");
-		}
+		/* Infinity loop workaround */
+		//while (!iommu_done) {
+			//print(".");
+		//}
 		print("\nIOMMU set\n");
 
 		/* Set the Device Table address for the TB stub to use */
-- 
2.25.1

