From 80533ea134c07045715d519b462da1e8495b1bbd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20Kotyla?= <michal.kotyla@3mdeb.com>
Date: Wed, 13 Apr 2022 16:40:07 +0200
Subject: [PATCH] Restore umask for grub.cfg

---
 util/grub-mkconfig.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/util/grub-mkconfig.in b/util/grub-mkconfig.in
index 9f477ff05..76764ad27 100644
--- a/util/grub-mkconfig.in
+++ b/util/grub-mkconfig.in
@@ -287,7 +287,10 @@ and /etc/grub.d/* files or please file a bug report with
     exit 1
   else
     # none of the children aborted with error, install the new grub.cfg
+    oldumask=$(umask)
+    umask 077
     mv -f ${grub_cfg}.new ${grub_cfg}
+    umask $oldumask
   fi
 fi
 
-- 
2.25.1
