From 4be25b0b35cf38d0dc3c3c6db674bfc36e6aa82f Mon Sep 17 00:00:00 2001
From: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>
Date: Wed, 13 Nov 2024 19:30:48 +0200
Subject: [PATCH] head: pass entry point address in %ebx to payload

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>
---
 head.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/head.S b/head.S
index 4c57eed..ba14ae4 100644
--- a/head.S
+++ b/head.S
@@ -246,7 +246,7 @@ GLOBAL(_entry)
 	 */
 	/* Linux expects Zero Page address in %esi, it is already there */
 	/* Multiboot2 expects MBI address in %ebx and magic number in %eax */
-	mov	%esi, %ebx
+	mov	%edi, %ebx
 	mov	$MULTIBOOT2_BOOTLOADER_MAGIC, %eax
 	/* Simple payload expects argument on stack followed by return address */
 	push	%esi
-- 
2.47.0
